config/load.go

```diff
@@ -0,0 +1,77 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/viper"
+)
+
+func Load(configFileDirs ...string) (*ArcConfig, error) {
+	arcConfig := getDefaultArcConfig()
+
+	err := setDefaults(arcConfig)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
If we add here this

```
	for _, configFileDir := range configFileDirs {
		viper.AddConfigPath(configFileDir)
	}
```

and the default value to "." on `cmd/arc/main.go:161`

```
configDir := flag.String("config", ".", "path to configuration yaml file")
```

Then I think we wouldn't need the change in `test/docker-compose.yml`, right? 🤔 

I think that may be better, because otherwise the `-config=...` parameter will be always required to start ARC and that may be a breaking change for operators of ARC and is not compatible with the instructions here: https://hub.docker.com/r/bsvb/arc/

Reply:
The beauty of having config in code is that we don't need the `config.yaml` at all. We can just run Arc without any config and it will run with the defaults. So no breaking changes here, except the fact that we can simplify that command and run in without passing any config to the docker (but I think it should stay that way to show how you can override the default config using docker)

There are 2 issues with having `.` as the default value for the config path:
1. The default config will always be overridden by the config in the root directory, even if the user didn't intentionally specify the `-config` flag.
2. If a user specifies the flag `-config`, but the path that he provided doesn't have any config file, we want to show user the error of e.g. `config not found in directory ...`.

The problem is that if we leave `.` as the default value, and there will be no config.yaml, because the user did not specify the `-config` flag, arc will return an error, because it will look for that config file. But Arc can start with no config.yaml (defaults are in code), so it shouldn't return an error when the flag is not passed.

One thing that this is possible to solve with `.` as the default value is to NOT inform the user when the config.yaml file does not exist and just run with the defaults when we can't find a config file. But I don't think that's a correct approach, because it may confuse the user a lot.

So, I decided that this semantic would be the best, having the current solution:
1. If the `-config` flag is not provided, assume user wants to use default config - don't search for any file in any directory.
2. If the `-config` flag is provided, assume user wants to override the default values with his values - look for that config file and throw an error is it can't be found.

In case of tests, the `-config=...` flag is needed, because tests have a config file that is a bit different from the defaults, that's why I added it to the docker-compose.